src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
 
-                CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> result = builder;
-                _pool.Free(builder);
-                return result;
+                // if we ran into any errors, don't give this item any special value.  It just has the value of our actual text.
+                var textString = text.ToString();
+                var valueString = indentationError != null ? textString : content.ToString();
+
+                var node = SyntaxFactory.InterpolatedStringText(
+                    SyntaxFactory.Literal(leading: null, textString, SyntaxKind.InterpolatedStringTextToken, valueString, trailing: null));
+
+                return indentationError != null
+                    ? node.WithDiagnosticsGreen(new[] { indentationError })
+                    : node;
             }
 
-            SyntaxToken getCloseQuote(Range openQuoteRange)
+            SyntaxToken getCloseQuote()
             {
                 // Make a token for the close quote " (even if it was missing)
-                var closeQuoteText = originalText[closeQuoteRange];
-                return closeQuoteText == ""
-                    ? SyntaxFactory.MissingToken(leading: null, SyntaxKind.InterpolatedStringEndToken, trailing: originalToken.GetTrailingTrivia())
-                    : SyntaxFactory.Token(leading: null, SyntaxKind.InterpolatedStringEndToken, closeQuoteText, closeQuoteText, originalToken.GetTrailingTrivia());
+                return TokenOrMissingToken(
+                    leading: null,
+                    kind switch
+                    {
+                        Lexer.InterpolatedStringKind.Normal => SyntaxKind.InterpolatedStringEndToken,
+                        Lexer.InterpolatedStringKind.Verbatim => SyntaxKind.InterpolatedStringEndToken,
+                        Lexer.InterpolatedStringKind.SingleLineRaw => SyntaxKind.InterpolatedSingleLineRawStringEndToken,
+                        Lexer.InterpolatedStringKind.MultiLineRaw => SyntaxKind.InterpolatedMultiLineRawStringEndToken,
+                        _ => throw ExceptionUtilities.UnexpectedValue(kind),
+                    },
+                    originalText[closeQuoteRange],
+                    originalToken.GetTrailingTrivia());
             }
         }
 
+        private static SyntaxToken TokenOrMissingToken(GreenNode? leading, SyntaxKind kind, string text, GreenNode? trailing)
+            => text == ""
+                ? SyntaxFactory.MissingToken(leading, kind, trailing)
+                : SyntaxFactory.Token(leading, kind, text, trailing);
+
+        private static int SkipWhitespace(ReadOnlySpan<char> text, int currentIndex)
+        {
+            while (currentIndex < text.Length && SyntaxFacts.IsWhitespace(text[currentIndex]))
+                currentIndex++;
+            return currentIndex;
+        }
+
+        private static int ConsumeRemainingContentOnLine(StringBuilder content, ReadOnlySpan<char> text, int currentIndex)
```

Comment:
>ConsumeRemainingContentOnLine [](http://example.com/codeflow?start=27&length=29)

nit: consider `ConsumeContentAndNewLine` to make it clear that we go past the newline character(s)

Reply:
done.